src/odemis/driver/xt_client.py

```diff
@@ -2271,6 +2281,55 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                         moving = self.parent.stage_is_moving()
                         if moving:
                             logging.warning("Stage reported stopped but moving again, will wait longer")
+                            continue
+
+                    # The stage is not moving anymore, but we still want to wait until the position has been updated
+                    check_pos = True
+                    # check for rotation and tilt is not included as it was not tested on the hardware
+                    # do not check update if angular movements are present
+                    if 'r' in pos or 't' in pos:
+                        logging.debug("Position includes rx or rz. Breaking out of the loop.")
+                        check_pos = False
+                    else:
+                        # only linear movements are requested, estimate time taken by stage
+                        # TODO check the linear speed, default to 2 (copied from move_to_pos in zeiss.py)
+                        pass
+                        # dur_lin = max(
+                        #     numpy.sqrt((pos - orig_pos_raw) ** 2)) / 2
+                        # # give enough time for the stage to update its position
+                        # timeout_update = dur_lin*10
+                    timeout_update = 10  # keep it to 10 seconds till the linear speed of stage is verified
+                    time_check = time.time()
+                    while check_pos:
+                        if time.time() > timeout_update + time_check:
+                            logging.warning(
+                                "Stage position after move + %s s is still not properly updated from original pos: "
+                                "%s. Giving up waiting", timeout_update, orig_pos_raw)
+
+                        moving = self.parent.stage_is_moving()
+                        if moving:
+                            logging.warning("Stage reported stopped but moving again, will wait longer")
+                            time.sleep(20e-3)
+                            continue
+
+                        pos_raw = self.parent.get_stage_position()
+                        # Every axis requested to move should have moved (compared to the position before starting)
+                        if all(pos_raw[a] != op for a, op in orig_pos_raw.items()):
+                            all_axes_updated = True
+
+                            # For x, y, and z, be extra picky and wait until they are "almost" at the target
+                            for k in ["x", "y", "z"]:
+                                if k in pos:
+                                    if abs(pos_raw[k] - target_pos[k]) >= 1e-6:
+                                        all_axes_updated = False
```

Comment:
We have a util to help with this check: ```odemis.util.driver.isNearPosition```
```suggestion
                            from odemis.util.driver import isNearPosition
                            # check if all axes have started moving
                            if not isNearPosition(orig_pos_raw, pos_raw, axes=set(orig_pos.raw.keys()):
                                # check if linear axes (x, y, z) are almost at target position
                                all_axes_updated =  isNearPosition(current_pos=pos_raw, target_position=target_pos, axes=["x", "y", "z"], atol_linear=1e-6)